DependencyInjection/Security/Factory/RefreshTokenAuthenticatorFactory.php

```diff
@@ -51,6 +52,10 @@ public function addConfiguration(NodeDefinition $builder): void
                 ->scalarNode('provider')->end()
                 ->scalarNode('success_handler')->end()
                 ->scalarNode('failure_handler')->end()
+                ->booleanNode('invalidate_token_on_logout')
+                    ->defaultFalse() // TODO - Enable by default in 2.0.
```

Comment:
> We should set it to null here and run a deprecation if the boolean has not been defined in order to activate it by default in 2.0 without BC.

I'd only do that if we needed the value to actually have a configuration (the same way the deprecation if you don't have the model class set in the bundle config works), because the 1.x branch supports both Symfony 4.4 and 5.4 we can't sanely keep the listener turned on by default for both workflows (which would result in the listener being registered on both the global event dispatcher and the firewall specific event dispatcher, and I have no idea how Symfony deals with merging and de-duping stuff for the firewall specific event dispatcher but I know it does move some stuff around).

> Isn't there a way to get the bundle configuration here or in the `createAuthenticator` in order to activate it by default when the 'logout_firewall' is null ? This would allow us to keep the default value on 'logout_firewall' without creating a BC and to launch a deprecation when it is defined.

We can get to container parameters, but we can't get the processed `$config` array.  For 1.x, I'm choosing to default to the listener being off by default with this backport and then restoring the current 1.1 behavior of it being active by default for 2.0 when we only have a single authentication flow to worry about.

As for the `logout_firewall` config node, that one's a bit of a pain point to work around.  It defaulting to "api" really should've been something added to the [Flex recipe](https://github.com/symfony/recipes-contrib/tree/main/gesdinet/jwt-refresh-token-bundle/1.0) for this bundle and not hardcoded into the bundle configuration.  So we can't rely on checking if its value matches the current default of "api", we don't know if the user explicitly wanted that or if it was implicitly set to that by way of not configuring it.  Changing its default to null gives us a better gauge of whether the node is explicitly configured or not and lets us do stuff with that information at compile time.